guacamole/src/main/webapp/app/form/directives/form.js

```diff
@@ -101,6 +133,46 @@ angular.module('form').directive('guacForm', [function form() {
 
             };
 
+            /**
+             * Determines whether the given object is a form, under the
+             * assumption that the object is either a form or a field.
+             *
+             * @param {Form|Field} obj
+             *     The object to test.
+             *
+             * @returns {Boolean}
+             *     true if the given object appears to be a form, false
+             *     otherwise.
+             */
+            var isForm = function isForm(obj) {
+                return !!('name' in obj && 'fields' in obj);
+            };
+
+            // Produce set of forms from any given content
+            $scope.$watch('content', function setContent(content) {
+
+                // If no content provided, there are no forms
+                if (!content) {
+                    $scope.forms = [];
+                    return;
+                }
+
+                // Ensure content is an array
+                if (!angular.isArray(content))
```

Comment:
> On Apr 16, 2015 5:06 PM, "James Muehlner" <james.muehlner@guac-dev.org> wrote:
> 
> That it's not an array
> 

Oh, that'll happen if ```content``` is given as a single ```Form``` or ```Field```, both of which are allowed in addition to arrays of the same.